src/components/CheckboxGroup/index.js

```diff
@@ -4,78 +4,80 @@ import classnames from 'classnames';
 import { uniqueId } from '../../libs/utils';
 import RenderIf from '../RenderIf';
 import RequiredAsterisk from '../Input/requiredAsterisk';
-import Checkbox from './checkbox';
+import CheckboxList from './checkboxList';
 import './styles.css';
 
 export default class CheckboxGroup extends Component {
     constructor(props) {
         super(props);
+        this.checkboxGruopId = uniqueId('checkboxGroup');
         this.handleOnChange = this.handleOnChange.bind(this);
-    }
-
-    getCheckboxValues() {
-        const { options } = this.props;
-        return options.map(option => (
-            <Checkbox
-                {...option}
-                isSelected={this.checkIfSelected(option)}
-                onChange={this.handleOnChange}
-                key={uniqueId('checkbox')} />
-        ));
+        this.checkIfSelected = this.checkIfSelected.bind(this);
     }
 
     getCheckboxContainerClassNames() {
-        const { error } = this.props;
-        return classnames('rainbow-checkbox-group-container', { 'rainbow-has-error': !!error });
+        const { error, className } = this.props;
+        return classnames(
+            'rainbow-checkbox-group-container',
+            { 'rainbow-checkbox-group-has-error': !!error },
+            className,
+        );
     }
 
-    handleOnChange(id, isSelected, disabled) {
-        if (disabled) return;
+    handleOnChange(event) {
+        const { value, checked } = event.target;
         const { value: values, onChange } = this.props;
-        if (!isSelected) {
-            onChange(values.filter(valueId => valueId !== id));
+        if (checked) {
+            onChange(values.concat([value]));
         } else {
-            onChange(values.concat([id]));
+            onChange(values.filter(valueId => valueId !== value));
         }
     }
 
     checkIfSelected(option) {
         const { value: values } = this.props;
-        return !!values.find(value => value === option.id);
+        return values.find(value => value === option.value) !== undefined;
     }
 
     render() {
-        const { required, label, error } = this.props;
+        const { options, required, label, error, style } = this.props;
         return (
-            <div className={this.getCheckboxContainerClassNames()}>
+            <fieldset className={this.getCheckboxContainerClassNames()} style={style}>
                 <RenderIf isTrue={!!label}>
-                    <legend className="rainbow-checkbox-group__label">
+                    <legend className="rainbow-checkbox-group-label">
                         <RequiredAsterisk required={required} />
                         {label}
                     </legend>
                 </RenderIf>
-                <div className="rainbow-checkbox-group_checkbox-container">
-                    {this.getCheckboxValues()}
+                <div id={this.checkboxGruopId} className="rainbow-checkbox-group-checkbox-container">
```

Comment:
this id is not here, is for the error container and only is passed if error exists